File: problems/2025/09-25/generate-binary-numbers/solution.js
class Solution {
  generateBinary(n) {
    const res = [];
    const q = ["1"];
    for (let i = 0; i < n; i++) {
      const s = q.shift();
      res.push(s);
      q.push(s + "0");
      q.push(s + "1");
    }
    return res;
  }
}

console.log(new Solution().generateBinary(4)); // ["1","10","11","100"]

File: problems/2025/09-25/generate-binary-numbers/Solution.java
import java.util.*;
public class Solution {
    public ArrayList<String> generateBinary(int n) {
        ArrayList<String> res = new ArrayList<>(n);
        Deque<String> q = new ArrayDeque<>();
        q.add("1");
        for (int i = 0; i < n; i++) {
            String s = q.removeFirst();
            res.add(s);
            q.addLast(s + "0");
            q.addLast(s + "1");
        }
        return res;
    }
    public static void main(String[] args) {
        System.out.println(new Solution().generateBinary(6));
    }
}
